### PR TITLE
implement fallback strategy for packages and make the accepted language modes more generic

### LINKED ISSUE #12: JSX
Hi! Thanks for this awesome project (and for building a bridge for  my transition away from VIM). You have done amazing work!

Would you be open to me adding support for the JSX language mode and perhaps a fallback mechanism that looks for an`index.jsx` if there is no `index.js`?

### PR DESCRIPTION
**Pull request by iammerrick**
Resolve #12 

This works via a fallback strategy which is great for polyglot codebases. :-)

### CODE REVIEW

File: lib/main.coffee [hunk 1]
```diff
@@ -2,6 +2,9 @@ path = require 'path'
 fs = require 'fs-plus'
 _ = require 'underscore-plus'
 
+hasCommonJSPackageSemantics = (scopeName, filePath) ->
+  fs.isDirectorySync(filePath) and (scopeName.startsWith('source.js') or scopeName.startsWith('source.coffee'))
+
```
Comment (t9md) [inline lib/main.coffee:7]: Any reason use `startsWith('source.js')` instead of `scopeName is "source.js"`?

Comment (iammerrick) [reply lib/main.coffee:7]: Yeah. The babel-language and other JavaScript language modes that add jsx sub-namespace the scopeName: `"source.js.jsx"`

Comment (t9md) [reply lib/main.coffee:7]: thanks! understood.

**Comment by t9md** [comment]
After I converted coffee to JS(for maintenance reason),  released as v0.5.0 now.

**Comment by iammerrick** [comment]
Thank you! This will be great to have!

**Pull request merged**